pandas/io/parsers/base_parser.py

```diff
@@ -341,6 +341,14 @@ def _extract_multi_indexer_columns(
         # extract the columns
         field_count = len(header[0])
 
+        # check if header lengths are equal
+        for header_iter in range(len(header)):
```

Comment:
You could do this like ``all(len(x) == len(ls[0]) for x in ls[1:])``

Reply:
So you do suggest removing that part of the error? It can simply say "Header rows must have an equal number of columns."

Reply:
Yeah I would go with that.